internal/services/oidc.go

```diff
@@ -106,33 +121,67 @@ func (osvc *OIDCService) IsFlowSecure(state string, token models.Tokens) (bool,
 	return true, nil
 }
 
-// CreateJWT creates a new token and the claims you would like it to contain.
-func (osvc *OIDCService) CreateJWT(cc *models.CustomClaims) (models.CustomClaims, error) {
-	// For HMAC signing method, the key can be any []byte
-	hmacRandSecret, err := randomByte(1990)
-	if err != nil {
-		osvc.logger.Errorf("An error occurred while generating HMAC. Error: %v", err)
-		return models.CustomClaims{}, err
-	}
+// Create a new JWT token based on Custom Claims models.
+func (osvc *OIDCService) Create(cc *models.CustomClaims) (models.CustomClaims, error) {
+	var err error
 
 	customClaims := models.CustomClaims{
 		Name:    cc.Name,
 		Email:   cc.Email,
 		Profile: cc.Profile,
 		RegisteredClaims: jwt.RegisteredClaims{
-			ExpiresAt: jwt.NewNumericDate(time.Now().Add(time.Duration(osvc.client.Configuration.JWTExpires))),
+			ExpiresAt: jwt.NewNumericDate(time.Now().Add(time.Duration(osvc.configuration.UserJWTExpires))),
 			IssuedAt:  jwt.NewNumericDate(time.Now()),
 		},
 	}
 
 	token := jwt.NewWithClaims(jwt.SigningMethodHS256, customClaims)
 
 	// Sign and get the complete encoded token as a string using the secret.
-	customClaims.AccessToken, err = token.SignedString(hmacRandSecret)
+	customClaims.AccessToken, err = token.SignedString(osvc.configuration.HMACSecret)
 
 	return customClaims, err
 }
 
+// Decode validates the token and returns the claims.
+func (osvc *OIDCService) Decode(tokenString string) (models.TokenInfo, error) {
+	token, err := jwt.Parse(tokenString, func(token *jwt.Token) (interface{}, error) {
+		// Don't forget to validate the alg is what you expect:
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+		}
+
+		return osvc.configuration.HMACSecret, nil
+	})
+
+	if token == nil {
+		return models.TokenInfo{}, errors.New("an unexpected error occurred while validating the JWT token")
+	}
+
+	if token.Valid {
+		claims := token.Claims.(jwt.MapClaims)
+		return models.TokenInfo{
+			UID:  fmt.Sprintf("%s%s", osvc.configuration.UserIDPrefix, claims[osvc.configuration.UserIDClaim].(string)),
+			Name: claims[osvc.configuration.UserNameClaim].(string),
+		}, nil
+	}
+
+	if ve, ok := err.(*jwt.ValidationError); ok {
+		if ve.Errors&jwt.ValidationErrorMalformed != 0 {
+			return models.TokenInfo{}, errors.New("that's not even a token")
+
+		} else if ve.Errors&jwt.ValidationErrorExpired != 0 {
+			return models.TokenInfo{}, authErrors.ErrTokenExpired
+		} else if ve.Errors&jwt.ValidationErrorNotValidYet != 0 {
+			return models.TokenInfo{}, authErrors.ErrTokenInactiveYet
+		}
+
+		return models.TokenInfo{}, fmt.Errorf("couldn't handle this token: %v", err)
+	}
```

Comment:
Can you remove the empty lines or add new where don't exist, just to be consistent and beautiful 😍.

Reply:
yeap, i can.